tokio-epoll-uring/src/system/lifecycle/handle.rs

```diff
@@ -171,4 +176,33 @@ impl crate::SystemHandle {
         let inner = self.inner.as_ref().unwrap();
         execute_op(op, inner.submit_side.weak(), None).await
     }
+
+    pub async fn statx<F: IoFd + Send>(
+        &self,
+        file: F,
+    ) -> (
+        F,
+        Result<
+            Box<uring_common::libc::statx>,
+            crate::system::submission::op_fut::Error<std::io::Error>,
+        >,
+    ) {
+        // TODO: avoid the allocation? allow callers to provide their own buffer?
+        let buf: Box<uring_common::libc::statx> = Box::new(
+            // TODO replace with Box<MaybeUninit>, https://github.com/rust-lang/rust/issues/63291
+            // SAFETY: we only use the memory if the fstat succeeds, should be using MaybeUninit here.
+            unsafe { std::mem::zeroed() },
+        );
```

Comment:
Not going to avoid allocations in this release.
I'd love to use a slab cache for this at some point.

But, generally, the cost of doing the syscall should be >> cost of the allocation.